### PR TITLE
KEP-769: Refactor wrapped_bzn_msg to bzn_envelope

### PR DESCRIPTION
**Pull request by isabelsavannah**

### CODE REVIEW

File: crypto/crypto.cpp [hunk 1]
```diff
@@ -35,8 +35,24 @@ crypto::crypto(std::shared_ptr<bzn::options_base> options)
     }
 }
 
+const std::string&
+crypto::extract_payload(const bzn_envelope& msg)
+{
+    switch (msg.payload_case())
+    {
+        case bzn_envelope::kPbftRequest : return msg.pbft_request();
```
Comment (ebruck) [inline crypto/crypto.cpp]: This is not our coding standard. Each return should be contained withing a block below "case"

File: crypto/crypto.cpp [hunk 2]
```diff
@@ -35,8 +35,24 @@ crypto::crypto(std::shared_ptr<bzn::options_base> options)
...
+        case bzn_envelope::kDatabaseResponse : return msg.database_response();
+        case bzn_envelope::kJson : return msg.json();
+        case bzn_envelope::kAudit : return msg.audit();
+        case bzn_envelope::kPbft : return msg.pbft();
+        case bzn_envelope::kPbftMembership : return msg.pbft_membership();
+        default : throw std::runtime_error(
```
Comment (ebruck) [inline crypto/crypto.cpp]: please reformat

File: crypto/crypto.hpp [hunk 3]
```diff
@@ -46,6 +46,8 @@ namespace bzn
 
         void log_openssl_errors();
 
+        const std::string& extract_payload(const bzn_envelope&);
```
Comment (ebruck) [inline crypto/crypto.hpp]: missing param variable

File: proto/bluzelle.proto [hunk 4]
```diff
@@ -18,31 +18,28 @@ import "database.proto";
...
 
+message bzn_envelope
+{
+    string sender = 1;
+    bytes signature = 2;
+
+    oneof payload
+    {
+        bytes pbft_request = 3;
+        bytes database_response = 4;
```
Comment (ebruck) [inline proto/bluzelle.proto]: Will a response ever arrive via this message type?

Comment (isabelsavannah) [reply proto/bluzelle.proto]: no, but it'll be sent out via it

**Pull request merged**